### ISSUE TITLE
Build fails in 3rdparty/minimp3/minimp3.h on arm64

### ISSUE THREAD

**Issue by popey** [mamedev/mame#5653]
I'm building MAME 0213, but I believe this has been broken since ~0210 (last successful build I have was tag g9df2514. It builds fine on x86_64, x86, armhf and even s390x, but not arm64 / aarch64.

Here's a snippet from the build log. 

```
Compiling src/devices/sound/k054539.cpp...
Compiling src/devices/sound/k056800.cpp...
Compiling src/devices/sound/l7a1045_l6028_dsp_a.cpp...
Compiling src/devices/sound/lc7535.cpp...
Compiling src/devices/sound/lmc1992.cpp...
Compiling src/devices/sound/mas3507d.cpp...
In file included from ../../../../../src/devices/sound/mas3507d.cpp:14:0:
../../../../../3rdparty/minimp3/minimp3.h:1275:5: error: "HAVE_SSE" is not defined [-Werror=undef]
 #if HAVE_SSE
     ^
../../../../../3rdparty/minimp3/minimp3.h:1477:5: error: "HAVE_SSE" is not defined [-Werror=undef]
 #if HAVE_SSE
     ^
cc1plus: all warnings being treated as errors
optional.make:21405: recipe for target 'obj/Release/src/devices/sound/mas3507d.o' failed
make[2]: *** [obj/Release/src/devices/sound/mas3507d.o] Error 1
make[1]: *** [optional] Error 2
Makefile:94: recipe for target 'optional' failed
make: *** [linux] Error 2
makefile:1300: recipe for target 'linux' failed
```

**Comment by cuavas** [comment]
You could change the problematic lines to `#if defined(HAVE_SSE)` since it never seems to define the macro if it isn't available (similarly for `HAVE_SIMD`).  Since this is a third-party component we'd prefer if you could get this fixed upstream.

**Comment by Tafoid** [comment]
Can you not use NOWERROR=1 in make to work around it for now?

**Comment by lieff** [comment]
Should be fixed in https://github.com/lieff/minimp3/commit/cf038fa3dfdb0f469fd23795939009bf3c354d4a

**Comment by jejones3141** [comment]
Alas, lieff/minimp3@cf038fa doesn't show up in the 0.244 sources.

**Comment by cuavas** [comment]
Well, someone actually needs to synchronise with upstream and test it, preferably someone familiar with the systems that depend on it.

**Comment by invertego** [comment]
This issue is long gone, as the fix picked up when minimp3 was updated in e6d3720985d2ac23c71ec85a552d11b6e27fa8e1.

Konami System 573 and Namco System 10 titles with MP3 audio are working in arm64 builds, too.
